SignalMessaging/attachments/SharingThreadPickerViewController.m

```diff
@@ -193,38 +263,30 @@ - (void)tryToSendAttachment:(SignalAttachment *)attachment fromViewController:(U
     }
 #endif
 
-    void (^sendCompletion)(NSError *_Nullable, TSOutgoingMessage *message) = ^(
-        NSError *_Nullable error, TSOutgoingMessage *message) {
-        AssertIsOnMainThread();
-
-        if (error) {
-            [fromViewController
-                dismissViewControllerAnimated:YES
-                                   completion:^(void) {
-                                       DDLogInfo(@"%@ Sending attachment failed with error: %@", self.logTag, error);
-                                       [self showSendFailureAlertWithError:error
-                                                                   message:message
-                                                        fromViewController:fromViewController];
-                                   }];
-            return;
-        }
+    SendCompletionBlock sendCompletion = ^(NSError *_Nullable error, TSOutgoingMessage *message) {
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (error) {
+                [fromViewController
+                    dismissViewControllerAnimated:YES
+                                       completion:^(void) {
+                                           DDLogInfo(@"%@ Sending message failed with error: %@", self.logTag, error);
+                                           [self showSendFailureAlertWithError:error
+                                                                       message:message
+                                                            fromViewController:fromViewController];
+                                       }];
+                return;
+            }
 
-        DDLogInfo(@"%@ Sending attachment succeeded.", self.logTag);
-        [self.shareViewDelegate shareViewWasCompleted];
+            DDLogInfo(@"%@ Sending message succeeded.", self.logTag);
+            [self.shareViewDelegate shareViewWasCompleted];
+        });
     };
 
     [fromViewController presentViewController:progressAlert
                                      animated:YES
                                    completion:^(void) {
-                                       __block TSOutgoingMessage *outgoingMessage =
-                                           [ThreadUtil sendMessageWithAttachment:self.attachment
-                                                                        inThread:self.thread
-                                                                   messageSender:self.messageSender
-                                                                      completion:^(NSError *_Nullable error) {
-                                                                          sendCompletion(error, outgoingMessage);
-                                                                      }];
-
-                                       self.outgoingMessage = outgoingMessage;
```

Comment:
It seems like we lost this assignment which is required to show upload progress.

Reply:
Ah, good catch.  Fixed and added a comment to make clear why this is necessary.